lib/SharedLib/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+# Runtime libs from tt-mlir
+set(TTNN_RUNTIME_LIBS TTRuntime TTRuntimeTTNN)
+
+# Dependency libs from tt-metal/ttnn project for ttnn runtime
+set(TTNN_LIBS TTMETAL_LIBRARY DEVICE_LIBRARY TTEAGER_LIBRARY TTNN_LIBRARY)
+
+# Libs from tt-mlir project
+set(TTMLIR_LIBS
+    TTNNTargetFlatbuffer
+    MLIRTTDialect
+    MLIRTTIRDialect
+    MLIRTTNNDialect
+    MLIRTTKernelDialect
+    TTMLIRTTIRToTTNN
+    MLIRTTMetalDialect
+    MLIRTTIRTransforms
+    MLIRTTNNTransforms
+    MLIRTTIRAnalysis
+    MLIRTTNNPipelines
+    TTMLIRTTNNToEmitC
+)
+
+# We supply empty.cpp because CMake does not allow creating a library without sources.
+add_library(TTMLIR SHARED empty.cpp)
+
+add_dependencies(TTMLIR
+    ${TTMLIR_LIBS}
+    ${TTNN_RUNTIME_LIBS}
+    ${TTNN_LIBS}
+)
+
+target_link_libraries(TTMLIR PRIVATE
```

Comment:
This should be `PUBLIC`, all of the symbols from TTMLIR_LIBS should be made available to front ends.  We can drop the `-Wl,--whole-archive` below.  See docs: https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents

Reply:
It shouldn't matter, `TTMLIR` target is a final executable, the static libs linked into it if marked `PUBLIC` means to export their symbols into the dylib.  I think there must be something somewhere else in the cmake tree that's messing with this.  Anyway, we can revisit, let's get this in as is :)